src/components/plots/rectanglePlot.ts

```diff
@@ -0,0 +1,110 @@
+        colorScale: Scale.AbstractScale<any, string>) {
+      super(xScale, yScale);
+
+      // Category scales should not have padding by default
+      if (xScale instanceof Scale.Ordinal) {
+        (<Scale.Ordinal> xScale).innerPadding(0).outerPadding(0);
+      }
+      if (yScale instanceof Scale.Ordinal) {
+        (<Scale.Ordinal> yScale).innerPadding(0).outerPadding(0);
+      }
+
+      this._colorScale = colorScale;
+      this.classed("rectangle-plot", true);
+    }
+
+    public addDataset(keyOrDataset: any, dataset?: any) {
+      super.addDataset(keyOrDataset, dataset);
+      return this;
+    }
+
+    protected _getDrawer(key: string) {
+      return new _Drawer.Rect(key, true);
+    }
+
+    /**
+     * @param {string} attrToSet One of ["x", "y", "fill"]. If "fill" is used,
+     * the data should return a valid CSS color.
+     */
+    public project(attrToSet: string, accessor: any, scale?: Scale.AbstractScale<any, any>) {
+      super.project(attrToSet, accessor, scale);
+      if (attrToSet === "fill") {
+        this._colorScale = this._projections["fill"].scale;
+      }
+      return this;
+    }
+
+    protected _generateAttrToProjector() {
+      var attrToProjector = super._generateAttrToProjector();
+
+      // Copy each of the different projectors
+      var x1Attr = attrToProjector["x"];
+      var y1Attr = attrToProjector["y"];
+      var x2Attr = attrToProjector["x2"];
+      var y2Attr = attrToProjector["y2"];
+
+      // Infer width, with an exception for ordinal scales
+      if (x2Attr === undefined) {
+        attrToProjector["width"] = () => (<Scale.Ordinal> this._xScale).rangeBand();
+      } else {
+        attrToProjector["width"] = (d, i, u, m) => {
+          return Math.abs(x2Attr(d, i, u, m) - x1Attr(d, i, u, m));
+        };
+      }
+
+      // Adjust x to respect the width for ordinal scales
+      attrToProjector["x"] = (d, i, u, m) => {
+        var offset = 0;
+        if (x2Attr === undefined) {
+          offset = attrToProjector["height"](d, i, u, m) / 2;
+        }
+        return x1Attr(d, i, u, m) - offset;
+      };
+
+
+      // Infer height, with an exception for ordinal scales
+      if (y2Attr === undefined) {
+        attrToProjector["height"] = () => (<Scale.Ordinal> this._yScale).rangeBand();
+      } else {
+        attrToProjector["height"] = (d, i, u, m) => {
+          return Math.abs(y2Attr(d, i, u, m) - y1Attr(d, i, u, m));
+        };
+      }
+
+      // Adjust y to respect the height, with an exception for ordinal scales
+      attrToProjector["y"] = (d, i, u, m) => {
+        var offset = attrToProjector["height"](d, i, u, m);
+        if (y2Attr === undefined) {
+          offset = offset / 2;
+        }
+        return y1Attr(d, i, u, m) - offset;
+      };
+
+      return attrToProjector;
+    }
+
+    protected _generateDrawSteps(): _Drawer.DrawStep[] {
+      return [{attrToProjector: this._generateAttrToProjector(), animator: this._getAnimator("cells")}];
+    }
```

Comment:
"cells" seems to be a copy paste remnant.  Overwrite with something appropriate?